src/utils/multicall.provider.ts

```diff
@@ -28,23 +27,30 @@ let multicall2Addresses: Record<number, string> = {
 
 const DEFAULT_BATCH_SIZE = 50;
 
-// return types considering T is a tuple
+// return types considering T could be a tuple
+type AllResult<T extends any[]> = [success: true, returns: T] | [success: false, returns: never[]];
 
-type AllResult<T> = [success: boolean, returns: T];
-
-type TryAllResult<T extends any[]> = {
-  [K in keyof T]: K extends number ? { success: boolean; returnData: T[K] } : T[K];
-};
+type TryAllResult<T extends any[]> = number extends T["length"]
+  ? {
+      [K in keyof T]: K extends number
+        ? { success: true; returnData: T[K] } | { success: false; returnData: never[] }
+        : T[K];
+    }
+  : {
+      [K in keyof T]: { success: true; returnData: T[K] } | { success: false; returnData: never[] };
+    };
 
-type GroupAllResult<T extends any[][]> = [
-  success: boolean,
-  returns: {
-    [K in keyof T]: T[K];
-  }
-];
+type GroupAllResult<T extends any[][]> =
+  | [
+      success: true,
+      returns: {
+        [K in keyof T]: T[K];
+      }
+    ]
+  | [success: false, returns: never[][]];
 
 type GroupTryAllResult<T extends any[][]> = {
-  [K in keyof T]: K extends number ? TryAllResult<T[K]> : T[K];
+  [K in keyof T]: TryAllResult<T[K]>;
```

Comment:
this change somehow fails my `npm test` 
I'm assuming it works for you both @RCantu92 @Vxatz ?

Reply:
this was fixed after updating the ts version in package.json